src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -2939,77 +2947,102 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
                 }
                 throw SemanticError("Only doc strings and const ellipsis allowed as expressions inside class", expr->base.base.loc);
             } else if( AST::is_a<AST::ClassDef_t>(*x.m_body[i]) ) {
-                visit_ClassDef(*AST::down_cast<AST::ClassDef_t>(x.m_body[i]));
+                visit_ClassDef(*AST::down_cast<AST::ClassDef_t>(x.m_body[i]));  
                 continue;
             } else if ( AST::is_a<AST::FunctionDef_t>(*x.m_body[i]) ) {
-                throw SemanticError("Struct member functions are not supported", x.m_body[i]->base.loc);
+                this->visit_stmt(*x.m_body[i]);
+                AST::FunctionDef_t* f_ast = AST::down_cast<AST::FunctionDef_t>(x.m_body[i]);
+                ASR::symbol_t* f_sym = current_scope->get_symbol(f_ast->m_name);
+                ASR::Function_t* f = ASR::down_cast<ASR::Function_t>(f_sym);
+                std::string class_proc_name = "Xx_Class_Procedure_"+std::string(f->m_name);
+                SymbolTable* proc_scope = ASRUtils::symbol_parent_symtab(f_sym);
+                Str s;
+                s.from_str(al,class_proc_name);
+                ASR::abiType abi = ASR::abiType::Source;
+                tmp = make_ClassProcedure_t(
+                    al,
+                    f_sym->base.loc,
+                    proc_scope,
```

Comment:
Here, Keep the ClassProcedure name as f_ast->name and Function name as a class_proc_name,
See my example: https://github.com/lcompilers/lpython/pull/2721#issuecomment-2145533534